packages/plugin-snowplow/src/androidMain/kotlin/ly.iterative.itly.snowplow/SnowplowOptions.kt

```diff
@@ -0,0 +1,14 @@
+/**
+ * Android SnowplowPlugin
+ */
+package ly.iterative.itly.snowplow
+
+import android.content.Context
+import com.snowplowanalytics.snowplow.controller.TrackerController
+
+actual open class SnowplowOptions(
+        val androidContext: Context,
+        var trackerUrl: String,
+        var vendor: String,
+        var tracker: TrackerController?
```

Comment:
I like how passing `tracker`  lets users configure their own tracker using Snowplows builders but it is kind of awkward that the user still needs to give a trackerUrl, vendor, androidContext. Not sure if there is a cleaner way to handle this. Any thoughts on a better way to pass in the options?

Reply:
@justin-fiedler Do you think having multiple constructors makes sense? Is that possible on multiplatform?

Reply:
@dantetam we could add multiple constructors to the SnowplowPlugin but we still need to expose all options via a single SegmentOptions class in codegen. [Example for the other destinations](https://bitbucket.org/seasyd/analytics_tools/src/5c695fbb3051aeb5ac0daa63161ecd85f1f0796f/sdk-examples/android-kotlin-v3/ItlyApp/app/src/main/java/com/example/itlyapp/ItlyKtActivity.kt#lines-37). I think for now we can make both `trackerUrl` and `tracker` optional and then verify in the SegmentPlugin constructor that at least one is set, else throw an error.